### PR TITLE
Add log_level, plural event_notification_configs to IoT registry

### PR DESCRIPTION
**Pull request by emilymye**
# Release Note for Downstream PRs (will be copied)
```releasenote
cloudiot: `resource_cloudiot_registry` now has fields plural `event_notification_configs` and `log_level`, and `event_notification_config` has been deprecated. 
```

### CODE REVIEW

File: third_party/terraform/resources/resource_cloudiot_registry.go [hunk 1]
```diff
@@ -369,3 +471,79 @@ func resourceCloudIoTRegistryStateImporter(d *schema.ResourceData, meta interfac
...
+	log.Printf("[DEBUG] State before upgrade: %#v", rawState)
+
+	oldCfg, ok := rawState["event_notification_config"]
+	if ok {
+		delete(rawState, "event_notification_config")
+	}
+
+	v, ok := rawState["event_notification_configs"]
+	if ok && v != nil {
+		// Just ignore old value if event_notification_configs is already set.
```
Comment (rileykarson) [inline third_party/terraform/resources/resource_cloudiot_registry.go]: Can the value already be set? As I understand, the upgrade will run during the first refresh upon using a new provider version?

File: third_party/terraform/resources/resource_cloudiot_registry.go [hunk 2]
```diff
@@ -369,3 +471,79 @@ func resourceCloudIoTRegistryStateImporter(d *schema.ResourceData, meta interfac
 	d.SetId(id)
 	return []*schema.ResourceData{d}, nil
 }
+
+func resourceCloudIotRegistryStateUpgradeV0toV1(rawState map[string]interface{}, meta interface{}) (map[string]interface{}, error) {
+	log.Printf("[DEBUG] State before upgrade: %#v", rawState)
+
+	oldCfg, ok := rawState["event_notification_config"]
+	if ok {
+		delete(rawState, "event_notification_config")
```
Comment (rileykarson) [inline third_party/terraform/resources/resource_cloudiot_registry.go]: What does deleting the value in state do? Won't we immediately add it back in `Read`?

File: third_party/terraform/tests/resource_cloudiot_registry_test.go [hunk 3]
```diff
@@ -93,6 +198,35 @@ func TestAccCloudIoTRegistry_update(t *testing.T) {
 	})
 }
 
+func TestAccCloudIoTRegistry_deprecatedEventNotificationConfig(t *testing.T) {
```
Comment (rileykarson) [inline third_party/terraform/tests/resource_cloudiot_registry_test.go]: Can you add a test where we do the opposite, using the new and then old field?

File: third_party/terraform/tests/resource_cloudiot_registry_test.go [hunk 4]
```diff
@@ -93,6 +118,64 @@ func TestAccCloudIoTRegistry_update(t *testing.T) {
...
+	registryName := fmt.Sprintf("tf-registry-test-%s", acctest.RandString(10))
+	topic := fmt.Sprintf("tf-registry-test-%s", acctest.RandString(10))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudIoTRegistryDestroy,
+		Steps: []resource.TestStep{
+			{
+				// Use deprecated field (event_notification_config) to create
```
Comment (rileykarson) [inline third_party/terraform/tests/resource_cloudiot_registry_test.go]:
```suggestion
				// Use new field (event_notification_config) to create
```

File: third_party/terraform/tests/resource_cloudiot_registry_test.go [hunk 5]
```diff
@@ -177,3 +262,49 @@ resource "google_cloudiot_registry" "foobar" {
...
+resource "google_pubsub_topic" "event-topic" {
+  name = "%s"
+}
+
+resource "google_cloudiot_registry" "foobar" {
+  depends_on = ["google_project_iam_binding.cloud-iot-iam-binding"]
+
+  name = "%s"
+
+  event_notification_config = {
```
Comment (pdecat) [inline third_party/terraform/tests/resource_cloudiot_registry_test.go]: Shouldn't have `event_notification_config` been replaced by its plural alternative?

**Pull request merged**